impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -492,6 +551,8 @@ public boolean replace(final K key, final V oldValue, final V newValue) throws C
   public void clear() throws CacheAccessException {
     try {
       map.clear();
+      //TODO: Is this thread safe ?
+      currentUsageinBytes.set(0l);
```

Comment:
It's not, but I wonder if the alternatives wouldn't be worse? We could iterate over all mappings and remove them one by one but I'm slightly worried about the performance impact.